layers/+completion/spacemacs-helm/packages.el

```diff
@@ -180,7 +180,10 @@
 
       ;; use helm by default for M-x, C-x C-f, and C-x b
       (unless (configuration-layer/package-usedp 'smex)
-        (global-set-key (kbd "M-x") 'helm-M-x))
+        (global-set-key (kbd "M-x") 'helm-M-x)
+        (spacemacs/set-leader-keys
+          "SPC" 'helm-M-x)
```

Comment:
Also, it is already done here: https://github.com/syl20bnr/spacemacs/blob/develop/layers/+completion/spacemacs-helm/packages.el#L219-L225